src/compiler/binder.ts

```diff
@@ -316,6 +315,8 @@ import {
 } from "./_namespaces/ts";
 import * as performance from "./_namespaces/ts.performance";
 
+import * as Debug from "./debug";

```

Comment:
Do we have a mechanism to infer a namespace import when performing auto-imports, or will we have to do this manually if we auto-import `Debug` for the first time in a new file?

Reply:
We have to do it manually, unfortunately; it will instead go and import this from the local `_namespaces/ts.ts` which is undesirable because you go through export helpers and such. I was going to write a lint for that but forgot about it until you mentioned it.

I don't really know what we can do to do better; I don't think we have in our current auto-import system a way to automatically detect that it's "preferred" to import specific modules via a namespace import. It would be nice to get something like this, as I'm sure it'd help someone who has a convention like this (yay dogfooding).

Reply:
The alternative is to just stop writing `Debug.assert` at all, and just write `import { assert } from "./debug"`. Honestly that might be a totally fine thing to do.

Reply:
Except that the NodeJS types are visible to the project, so it's possible we'd inadvertently import the wrong `assert` via auto-imports.

Reply:
I _think_ we would actually get the local import first. (That and the compiler doesn't have node types at all, which helps.)

Reply:
> (That and the compiler doesn't have node types at all, which helps.)

Except they do show up for auto-imports, because the node types are present in node_modules:

![image](https://user-images.githubusercontent.com/3902892/217078548-7d8e2e0c-2ffe-49c1-b741-e129964e6a59.png)

And we don't seem to prioritize the local import.

Reply:
Ouch.

Reply:
Just to note it here, @andrewbranch has been working on doing some analysis of existing imports such that this sort of thing would work out of the box. I think we were going to go forward with this anyway without that feature (per a design meeting a while ago), but, it's great that we're going to improve it.